### PR TITLE
Handle the case where the domain or header domain is malformed

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull requests modifies the code to handle the case where the domain or header domain is malformed.

## 💭 Motivation and context ##

There was a case this past weekend where an HHS header domain was malformed such that the `get_public_suffix()` call failed because it was being passed a domain of `None`.  With these changes the code will handle such cases without crashing and will print out a helpful logging message.

## 🧪 Testing ##

All `pre-commit` hooks pass.  I also used these changes to successfully generate the HHS report from this past weekend.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - _eschew scope creep!_
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

## ✅ Post-merge checklist ##

- [x] Add a tag.

### CODE REVIEW

**Review by dav3r — APPROVED** [review]
Good sleuthing, nice fix and cleanup! 🔎 👍

File: src/report/generate_trustymail_report.py [hunk 1]
```diff
@@ -1130,13 +1130,28 @@ def process_record(domain, record, policy_published):
                                 # DKIM alignment is relaxed, so the header and
                                 # the domain just need to come from the same
                                 # base domain
-                                base_domain = self.__psl.get_public_suffix(
-                                    y["domain"]
-                                ).lower()
+                                domain = y["domain"]
+                                base_domain = self.__psl.get_public_suffix(domain)
+                                if base_domain is None:
+                                    logging.warn(
```
Comment (mcdonnnj) [inline src/report/generate_trustymail_report.py]:
```suggestion
                                    logging.warning(
```

File: src/report/generate_trustymail_report.py [hunk 2]
```diff
@@ -1130,13 +1130,28 @@ def process_record(domain, record, policy_published):
...
+                                    results.append("unaligned")
+                                    continue
+
                                 header_base_domain = self.__psl.get_public_suffix(
                                     header_from
-                                ).lower()
-                                if base_domain == header_base_domain:
+                                )
+                                if header_base_domain is None:
+                                    logging.warn(
```
Comment (mcdonnnj) [inline src/report/generate_trustymail_report.py]:
```suggestion
                                    logging.warning(
```

File: src/report/generate_trustymail_report.py [hunk 3]
```diff
@@ -1181,13 +1196,28 @@ def process_record(domain, record, policy_published):
                                 # SPF alignment is relaxed, so the header and
                                 # the domain just need to come from the same
                                 # base domain
-                                base_domain = self.__psl.get_public_suffix(
-                                    y["domain"]
-                                ).lower()
+                                domain = y["domain"]
+                                base_domain = self.__psl.get_public_suffix(domain)
+                                if base_domain is None:
+                                    logging.warn(
```
Comment (mcdonnnj) [inline src/report/generate_trustymail_report.py]:
```suggestion
                                    logging.warning(
```

File: src/report/generate_trustymail_report.py [hunk 4]
```diff
@@ -1181,13 +1196,28 @@ def process_record(domain, record, policy_published):
...
+                                    results.append("unaligned")
+                                    continue
+
                                 header_base_domain = self.__psl.get_public_suffix(
                                     header_from
-                                ).lower()
-                                if base_domain == header_base_domain:
+                                )
+                                if header_base_domain is None:
+                                    logging.warn(
```
Comment (mcdonnnj) [inline src/report/generate_trustymail_report.py]:
```suggestion
                                    logging.warning(
```

**Review by mcdonnnj — CHANGES_REQUESTED** [review]
Makes sense, but from the [Python `logging` documentation](https://docs.python.org/3/library/logging.html#logging.Logger.warning):

> Note
>
> There is an obsolete method `warn` which is functionally identical to `warning`. As `warn` is deprecated, please do not use it - use `warning` instead.

**Review by mcdonnnj — APPROVED** [review]

**Review by dav3r — APPROVED** [review]
👍 👍

**Review by mcdonnnj — APPROVED** [review]
`:=` :godmode:

**Pull request merged**